### PR TITLE
Fix faucet transaction

### PR DESCRIPTION
**Pull request by PawelKrawczyk96**

### CODE REVIEW

File: src/store/sagas/wallet.ts [hunk 1]
```diff
@@ -116,28 +118,47 @@ export function* handleAirdrop(): Generator {
...
-  const tokensAccounts = yield* select(accounts)
-  const collateralTokenProgram = yield* call(getToken, collateralTokenAddress)
-  let accountAddress = tokensAccounts[collateralTokenAddress.toString()]
-    ? tokensAccounts[collateralTokenAddress.toString()].address
-    : null
-  if (accountAddress == null) {
-    accountAddress = yield* call(createAccount, collateralTokenProgram.publicKey)
+  const instructions: Transaction[] = []
+  for (const [index, collateral] of collateralsAddresses.entries()) {
+    yield* call(setEmptyAccounts, collateralsAddresses)
```
Comment (awojciak) [inline src/store/sagas/wallet.ts:149]: why you call this every time in the loop?

File: src/store/sagas/wallet.ts [hunk 2]
```diff
@@ -116,28 +118,47 @@ export function* handleAirdrop(): Generator {
...
+    const tokensAccounts = yield* select(accounts)
```
Comment (awojciak) [inline src/store/sagas/wallet.ts:150]: move this also from the loop

File: src/store/sagas/wallet.ts [hunk 3]
```diff
@@ -116,28 +118,47 @@ export function* handleAirdrop(): Generator {
...
+    const ix = Token.createMintToInstruction(
+      TOKEN_PROGRAM_ID,
+      collateral,
+      tokensAccounts[collateral.toString()].address,
+      airdropAdmin.publicKey,
+      [],
+      collateralsQuantities[index]
+    )
+    instructions.push(new Transaction().add(ix))
```
Comment (awojciak) [inline src/store/sagas/wallet.ts:159]: just push ix

File: src/store/sagas/wallet.ts [hunk 4]
```diff
@@ -178,35 +199,43 @@ export function* signAndSend(wallet: WalletAdapter, tx: Transaction): SagaGenera
   const signature = yield* call([connection, connection.sendRawTransaction], signedTx.serialize())
   return signature
 }
-export function* createAccount(tokenAddress: PublicKey): SagaGenerator<PublicKey> {
+export function* createAccount(tokenAddress: PublicKey[]): SagaGenerator<PublicKey[]> {
```
Comment (awojciak) [inline src/store/sagas/wallet.ts:202]: we're using this saga also in swap and positions, so there might be types errors

**Review by awojciak — CHANGES_REQUESTED** [review]
i've left comments

**Review by wojciech-cichocki — CHANGES_REQUESTED** [review]
I leave comments from myself

File: src/store/sagas/wallet.ts [hunk 5]
```diff
@@ -178,35 +199,43 @@ export function* signAndSend(wallet: WalletAdapter, tx: Transaction): SagaGenera
...
+    const token = yield* call(getTokenDetails, tokenAddress[index].toString())
+    yield* put(
+      actions.addTokenAccount({
+        programId: address,
+        balance: new BN(0),
+        address: associatedAccs[index],
+        decimals: token.decimals
+      })
+    )
+    yield* call(sleep, 1000)
```
[the remark this answers is not in the document]
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:236]: done

File: src/store/sagas/wallet.ts [hunk 6]
```diff
@@ -105,9 +105,11 @@ export function* handleAirdrop(): Generator {
       }
     }
   }
-  yield* call(getCollateralTokenAirdrop, USDC.address, 100 * 10 ** USDC.decimal)
-  yield* call(getCollateralTokenAirdrop, USDT.address, 100 * 10 ** USDT.decimal)
-  yield* call(getCollateralTokenAirdrop, SOL.address, 10 ** SOL.decimal)
+  const collateralsAddresses: PublicKey[] = [USDC.address, USDT.address, SOL.address]
```
[the remark this answers is not in the document]
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:108]: done

File: src/store/sagas/wallet.ts [hunk 7]
```diff
@@ -116,28 +118,47 @@ export function* handleAirdrop(): Generator {
...
+    const accountAddress = tokensAccounts[collateral.toString()]
+      ? tokensAccounts[collateral.toString()].address
+      : null
+    if (accountAddress == null) {
+      acc.push(collateralTokenProgram.publicKey)
+    }
+  }
+  if (acc.length !== 0) {
+    yield* call(createAccount, acc)
+  } else {
```
[the remark this answers is not in the document]
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:138]: I'll remove this. It's not necessary

File: src/store/sagas/wallet.ts (hunk 3 above)
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:159]: Okay i'll notice that.

File: src/store/sagas/wallet.ts (hunk 4 above)
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:202]: done, i created second function to create multiple accounts

File: src/store/sagas/wallet.ts (hunk 1 above)
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:149]: done

File: src/store/sagas/wallet.ts (hunk 2 above)
Comment (PawelKrawczyk96) [reply src/store/sagas/wallet.ts:150]: done

**Review by awojciak — APPROVED** [review]

**Pull request merged**